RNA-seq/05-nb_cell_line_DESeq2.Rmd

```diff
@@ -154,124 +169,131 @@ Let's save this to our results file.
+                            # We will use `ashr` for estimation
+                            type = "ashr"
+                            )
 ```
 
-We need to extract the normalized gene matrix from the `vst_data` object. 
+Let's compare the log2 fold change estimates from from the two results tables:
 
-```{r vst-assay}
-vst_data <- assay(vst_data)
-```
+First we will combine the results into a new data frame.
 
-We will make a heatmap of only the significant genes, so let's obtain the gene IDs that have an adjusted p-value less than 0.05.
+```{r compare_shrink}
+comparison_df <- data.frame(
+  lfc_original = deseq_results$log2FoldChange,
+  lfc_shrunken = deseq_shrunken$log2FoldChange,
+  logmean = log10(deseq_results$baseMean)
+  )
+```
 
-```{r signif_genes, live = TRUE}
-# Filter to padj < 0.05 and then only extract the Genes
-signif_genes <- deseq_df %>%
-  dplyr::filter(padj < 0.05) %>%
-  dplyr::pull(Gene)
+Now we can plot the original and shrunken log2 fold change values to see what happened after shrinkage.
+
+```{r plot_comparison}
+ggplot(comparison_df, 
+       aes(x = lfc_original, 
+           y = lfc_shrunken, 
+           color = logmean)) +
+  geom_point(alpha = 0.1) +
+  theme_bw() +
+  scale_color_viridis_c() + 
+  coord_cartesian(xlim = c(-10,10), ylim = c(-10,10)) # zoom in on the middle
 ```
 
-Now we will use `signif_genes` to filter `vst_data` to only have significant genes' data. 
+We will now do a bit of manipulation to store the results in a data frame for easier manipulation.
 
-```{r signif_vst_data}
-signif_vst_data <- vst_data %>%
+```{r results_dataframe}
+# this is of class DESeqResults -- we want a data frame
+deseq_df <- deseq_shrunken %>%
+  # convert to a data frame
   as.data.frame() %>%
-  dplyr::filter(rownames(vst_data) %in% signif_genes)
+  # the gene ids were stored as row names -- let's them a column for easy display
+  tibble::rownames_to_column(var = "gene_id") %>%
+  # add on the gene names from the original deseq object
+  dplyr::mutate(gene_name = rowData(deseq_object)$gene_name)
 ```
 
-Let's make the results into a heatmap using the `pheatmap` package. 
-We will use the `scale = "row"` argument so that the data is standardized by gene (i.e., will be made into [z-scores](https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/z-score/)). 
-Standardizing the data by gene makes it so our heatmap is more easily interpretable despite how different genes' distributions may be. 
-
-```{r pheatmap}
-pheatmap::pheatmap(signif_vst_data, 
-                   show_rownames = FALSE,
-                   scale = "row",
-                   colorRampPalette(c("deepskyblue",
-                                      "black",
-                                      "yellow"))(25))
+Let's print out the results table, sorted by log2 fold change. 
+The highest values should be genes more expressed in the MYCN amplified cell lines.
+
+```{r sorted_table, live = TRUE}
```

Comment:
I think the answer to this is going to be no, but I will go ahead and ask anyway. Do we have control of the column name `gene_name`? Ideally that would say `gene_symbol` instead.

Reply:
Of course we do, we made it! I mean, we added it from the tximeta data, were it was called gene_name, but we don't have to keep that in _our_ data frame. Will fix.

Reply:
One word of caution though: folks will need to create the `SummarizedExperiment` themselves during a guided exercise, so could be a point of friction.

Reply:
The change I made for this is at:

https://github.com/AlexsLemonade/training-modules/blob/0d549abbaad60b3954bbc94be1ddc59be5efdc1e/RNA-seq/05-nb_cell_line_DESeq2.Rmd#L251

So the SummarizedExperiment is unchanged, just the column name when we add it into our results data frame. I don't think that is too bad?

Reply:
Sounds good 👍